src/datachain/query/dataset.py

```diff
@@ -444,30 +457,32 @@ def populate_udf_table(self, udf_table: "Table", query: Select) -> None:
                             "distributed processing."
                         )
 
-                    from datachain.catalog.loader import get_distributed_class
-
-                    distributor = get_distributed_class(
-                        min_task_size=self.min_task_size
-                    )
-                    distributor(
-                        self.udf,
-                        catalog,
-                        udf_table,
-                        query,
-                        workers,
-                        processes,
+                    from datachain.catalog.loader import get_udf_distributor_class
+
+                    udf_distributor_class = get_udf_distributor_class()
+                    udf_distributor = udf_distributor_class(
+                        catalog=catalog,
+                        table=udf_table,
+                        query=query,
+                        udf_data=filtered_cloudpickle_dumps(self.udf),
+                        batching=batching,
+                        workers=workers,
+                        processes=processes,
                         udf_fields=udf_fields,
+                        use_cache=self.cache,
                         is_generator=self.is_generator,
-                        use_partitioning=use_partitioning,
-                        cache=self.cache,
+                        min_task_size=self.min_task_size,
+                        rows_total=rows_total,
                     )
```

Comment:
Here I am improving params passed to UDF distributor class. See related SaaS PR.